lib/rspec/sidekiq/matchers/base.rb

```diff
@@ -313,24 +313,25 @@ def count_message
         end
 
         def failure_message_when_negated
-          message = ["expected not to #{common_message} but enqueued #{actual_jobs.count}"]
-          message << "  arguments: #{expected_arguments}" if expected_arguments.any?
-          message << "  options: #{expected_options}" if expected_options.any?
-          message.join("\n")
+          ["expected not to #{common_message} but enqueued #{actual_jobs.count}"].tap do |message|
+            message << "  arguments: #{expected_arguments}" if expected_arguments.any?
+            message << "  options: #{expected_options}" if expected_options.any?
+          end.join("\n")
         end
 
         def formatted(thing)
           RSpec::Support::ObjectFormatter.format(thing)
         end
 
         def normalize_arguments(args)
-          if args.is_a?(Array)
+          case args
+          when Array
             args.map{ |x| normalize_arguments(x) }
-          elsif args.is_a?(Hash)
+          when Hash
             args.each_with_object({}) do |(key, value), hash|
               hash[key.to_s] = normalize_arguments(value)
             end
-          elsif args.is_a?(Symbol)
+          when Symbol
```

Comment:
🎨 This section isn't impacted by frozen string literals, the node warnings, nor leaky spec constant so this change seems like it's just a preference. If so, I'd prefer we leave it unaltered if the `if ... elsif .. else` style with `is_a?`.

As an aside, since `case` statements go through a cascading set of pattern matching, the direct `if ... elsif` check against class is _slightly_ more performant in most cases. Testing with an array of a few million integers so we go through all the cases to the `else` each time:

```
Rehearsal --------------------------------------------
if is_a?   0.179518   0.005564   0.185082 (  0.185659)
case       0.188583   0.003112   0.191695 (  0.192592)
----------------------------------- total: 0.376777sec

               user     system      total        real
if is_a?   0.178338   0.003004   0.181342 (  0.182129)
case       0.207480   0.001087   0.208567 (  0.209595)
```